var/spack/repos/builtin/packages/mvapich2/package.py

```diff
@@ -347,11 +348,40 @@ def setup_dependent_build_environment(self, env, dependent_spec):
         env.set('MPICH_F90', spack_fc)
         env.set('MPICH_FC', spack_fc)
 
-    def setup_compiler_environment(self, env):
-        # For Cray MPIs, the regular compiler wrappers *are* the MPI wrappers.
-        # Cray MPIs always have cray in the module name, e.g. "cray-mvapich"
+    def _decide_if_external_is_cray(self):
+        '''For Cray MPIs, the regular compiler wrappers *are* the MPI wrappers.
+           Cray MPIs always have cray in the module name, e.g. "cray-mvapich"
+           make no assumption about where the module may be in the module list.
+
+           inspect external modules and:
+               - return True if we believe it is Cray
+               - return False otherwise
+        '''
         external_modules = self.spec.external_modules
-        if external_modules and 'cray' in external_modules[0]:
+        expected_prefix = 'cray-mvapich'
+        have_cray = False
+        if external_modules:
+            tty.msg('Detected external modules, will test for a'
+                    ' module that starts with {0}'.format(expected_prefix))
+            for module_name in external_modules:
+                if module_name.startswith(expected_prefix):
+                    have_cray = True
+                    msg = 'Found module with name {0} - assuming Cray environment'
+                    tty.msg(msg.format(module_name))
+        return have_cray
+
+    def setup_compiler_environment(self, env):
+        have_cray = self._decide_if_external_is_cray()
+
+        # use the this (since it's used elsewhere) to tag a package's message
+        pre = '{s.name}@{s.version} :'.format(s=self.spec)
+        if have_cray:
+            tty.msg(pre + 'Assuming Cray enviroment and setting MPI ENV'
+                    ' variables to spack wrappers')
+            tty.verbose(pre + 'Setting MPICC={0}'.format(spack_cc))
+            tty.verbose(pre + 'Setting MPICXX={0}'.format(spack_cxx))
+            tty.verbose(pre + 'Setting MPIF77={0}'.format(spack_fc))
+            tty.verbose(pre + 'Setting MPIF90={0}'.format(spack_fc))
```

Comment:
This should be less verbose once you're done with testing it.

Reply:
Is it possible to have them become `debug`, e.g., `spack -d` (is there a `tty.debug`)- I think `--verbose` should show it in the current incarnation.

I'm pushing back a little, because these messages were added because this is a deviation from the norm. It seems a bit redundant to list all wrappers, but if it only showed with `-d` or `-dd` that would enable really good info for bug reports.

To sort out what was going on, I ended up putting `traceback` call in the spec.py files - then I knew where these variables got set - my intuition blamed Cray or the modules - but it turned out to be a packaging thing.

Reply:
I've changed all output to be `tty.debug` - this keeps the console clean by default.

Reply:
I've changed to `tty.debug` which keeps the console clean by default.

For the setup routine, you can't read `env` so I had to do a little extra leg work. The code organization allows for printing what spack does.